src/routes/api/users.js

```diff
@@ -1,87 +1,83 @@
-            }
-            if (typeof req.body.user.email !== "undefined") {
-                user.email = req.body.user.email;
-            }
-            if (typeof req.body.user.bio !== "undefined") {
-                user.bio = req.body.user.bio;
-            }
-            if (typeof req.body.user.image !== "undefined") {
-                user.image = req.body.user.image;
-            }
-            if (typeof req.body.user.password !== "undefined") {
-                user.setPassword(req.body.user.password);
-            }
+      // only update fields that were actually passed...
+      if (typeof req.body.user.username !== 'undefined') {
+        user.username = req.body.user.username;
+      }
+      if (typeof req.body.user.email !== 'undefined') {
+        user.email = req.body.user.email;
+      }
+      if (typeof req.body.user.bio !== 'undefined') {
+        user.bio = req.body.user.bio;
+      }
+      if (typeof req.body.user.image !== 'undefined') {
+        user.image = req.body.user.image;
+      }
+      if (typeof req.body.user.password !== 'undefined') {
+        user.setPassword(req.body.user.password);
+      }
 
-            return user.save().then(function() {
-                return res.json({ user: user.toAuthJSON() });
-            });
-        })
-        .catch(next);
+      return user.save().then(() => res.json({ user: user.toAuthJSON() }));
+    })
+    .catch(next);
 });
 
-router.post("/users/login", function(req, res, next) {
-    if (!req.body.user.email) {
-        return res.status(422).json({ errors: { email: "can't be blank" } });
-    }
+router.post('/users/login', (req, res, next) => {
+  if (!req.body.user.email) {
+    return res.status(422).json({ errors: { email: "can't be blank" } });
+  }
 
-    if (!req.body.user.password) {
-        return res.status(422).json({ errors: { password: "can't be blank" } });
+  if (!req.body.user.password) {
+    return res.status(422).json({ errors: { password: "can't be blank" } });
+  }
+  passport.authenticate('local', { session: false }, (
+    err,
+    user,
+    info
+  ) => {
+    if (err) {
+      return next(err);
     }
-    passport.authenticate("local", { session: false }, function(
-        err,
-        user,
-        info
-    ) {
-        if (err) {
-            return next(err);
-        }
 
-        if (user) {
-            return res.json({ user: user.toAuthJSON() });
-        } else {
-            return res.status(422).json(info);
-        }
-    })(req, res, next);
+    if (user) {
+      return res.json({ user: user.toAuthJSON() });
+    }
+    return res.status(422).json(info);
+  })(req, res, next);
 });
 
-router.post("/users", function(req, res, next) {
-    const user = new User();
+router.post('/users', (req, res, next) => {
+  const user = new User();
 
-    user.username = req.body.user.username;
-    user.email = req.body.user.email;
-    user.setPassword(req.body.user.password);
+  user.username = req.body.user.username;
```

Comment:
Nice work here Funmi. But I'm wondering, is there a way we can make this and the next few lines cleaner through destructuring?

Reply:
@darasimiolaifa  i don't think that is my code.

Reply:
@darasimiolaifa  i think that is the default code that came with the repository, we will edit that as time goes on, its not related to her task.

Reply:
Okay. My bad. I saw that a lot of changes occurred and those were not taken care of, that was why. @funmi5 @oyewoas. Well done guys.